Data/HashMap/Internal.hs

```diff
@@ -2322,10 +2322,16 @@ searchSwap mary n toFind start = go start toFind start
 -- @
 -- xs ``disjoint`` ys = null (xs ``intersection`` ys)
 -- @
+--
+-- @since FIXME
 disjoint :: Eq k => HashMap k a -> HashMap k b -> Bool
 disjoint = disjointSubtrees 0
 {-# INLINE disjoint #-}
 
+-- Note that as of GHC 9.12, SpecConstr creates a specialized worker for
+-- handling the Collision vs. {BitmapIndexed,Full} and vice-versa cases,
+-- but this worker fails to be properly specialized for different key
+-- types. See https://gitlab.haskell.org/ghc/ghc/-/issues/26615.
```

Comment:
Should we perform constructor specialization manually to work around this?

Reply:
I don't mind if you or someone else wants to work on this, but I won't.